mx_benchmark.py

```diff
@@ -0,0 +1,304 @@
+#
+# Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+# or visit www.oracle.com if you need additional information or have any
+# questions.
+#
+# ----------------------------------------------------------------------------------------------------
+
+import json
+import re
+import socket
+import subprocess
+import uuid
+from argparse import ArgumentParser
+
+import mx
+
+
+_benchmark_suites = {}
+
+
+class BenchmarkSuite(object):
+    def name(self):
+        """Returns the name of the suite.
+
+        Returns:
+            basestring: Name of the suite.
+        """
+        raise NotImplementedError()
+
+    def benchmarks(self):
+        """Returns the list of the benchmarks provided by this suite.
+
+        Returns:
+            list of strings: List of benchmarks.
+        """
+        raise NotImplementedError()
+
+    def extraDimensions(self, benchmark, vmargs, runargs):
+        """Returns dictionary of extra dimensions for the given benchmark and arguments.
+
+        Arguments:
+            benchmark (string): Name of the benchmark.
+            vmargs (list of strings): List of VM arguments.
+            runargs (list of strings): List of benchmark arguments.
+
+        Returns:
+            dict[basestring, T]: Dictionary of field-name/value pairs.
+        """
+        raise NotImplementedError()
+
+    def run(self, benchmarks, vmargs, runargs):
+        """Runs the specified benchmarks with the given arguments.
+
+        Arguments:
+            benchmarks (list of strings): List of benchmark names.
+            vmargs (list of strings): List of VM arguments.
+            runargs (list of strings): List of benchmark arguments.
+
+        Returns:
+            object: A list of measurement results.
+
+            A measurement result is an object that can be converted into JSON and is
+            merged with root.
+
+            A measurement result must contain a field `metric`, which has the following
+            values:
+                - `metric.name` -- name of the metric (e.g. `"time"`, `"memory"`, ...)
+                - `metric.value` -- value of the measurement (e.g. `1.54`)
+                - `metric.unit` -- a string that specified the unit of measurement
+                  (e.g. `"ms"`)
+                - `metric.score-function` -- name of the scoring function for the result
+                  (e.g. `id`, which just returns the measurement value)
+                - `metric.score-value` -- score of the value, evaluated by the specified
+                  scoring function (e.g. `"1.54"`)
+                - `metric.type` -- a string denoting the type of the metric
+                  (e.g. `"numeric"`)
+                - `metric.better` -- `"higher"` if higher is better, `"lower"` otherwise
+                - `metric.iteration` -- iteration number of the measurement (e.g. `0`)
+        """
+        raise NotImplementedError()
+
+
+def add_benchmark_suite(suite):
```

Comment:
Maybe we should adopt the naming convention of `bmsuite` for BenchmarkSuite typed variables to minimize the confusion with normal mx suites.